synapse/handlers/room_member.py

```diff
@@ -178,62 +178,6 @@ async def ratelimit_invite(
 
         await self._invites_per_user_limiter.ratelimit(requester, invitee_user_id)
 
-    async def _can_join_without_invite(
-        self, state_ids: StateMap[str], room_version: RoomVersion, user_id: str
-    ) -> bool:
```

Comment:
This is moved to a separate handler to avoid an import loop between the `RoomMemberHandler` and `FederationHandler`.